server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -872,6 +873,17 @@ public void writeVIntArray(int[] values) throws IOException {
         }
     }
 
+    public void writeBigIntArray(IntArray values) throws IOException {
+        // TODO do we need to specialize this for ReleasableIntArray
+        if (values.size() > Integer.MAX_VALUE / Integer.BYTES) {
+            throw new UnsupportedOperationException(); // NOCOMMIT What do?
+        }
+        writeVInt((int) values.size() * Integer.BYTES);
+        for (int i = 0; i < values.size(); i++) {
```

Comment:
This seems needlessly expensive when we already have the byte representation for the implementation above. Should we maybe add a `writeTo` to `IntArray` (i.e. make it writable) instead to be able to leverage those bytes?

Reply:
Yeah!